### PR TITLE
Add New-ServiceClient utility function in generated module to enable mock testing

### LINKED ISSUE #298: Add New-ServiceClient utility function in generated module to enable mock testing

### PR DESCRIPTION
**Pull request by bmanikm**
Resolves #298.

### CODE REVIEW

File: PSSwagger/New-ServiceClient.ps1 [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Microsoft.PowerShell.Core\Set-StrictMode -Version Latest
```
Comment (brywang-msft) [inline PSSwagger/New-ServiceClient.ps1]: Any reason why you don't want to add to GeneratedHelpers?

Comment (bmanikm) [reply PSSwagger/New-ServiceClient.ps1]: Module owner can just replace this file during the mock testing.

**Comment by brywang-msft** [comment]
Start a readme/docs section for testing generated modules and include this strategy there.

**Comment by brywang-msft** [comment]
Also going to let Jeffrey sign off on this one. Approving from a code perspective.

**Comment by bmanikm** [comment]
I discussed with Jeffrey on this, it looks like he also need another utility function to allow the module owner to override the ServiceCredentials, SubscriptionId, BaseUri and HttpClientHandler properties of the service client object.

**Comment by deathly809** [comment]
The only suggestion I had was that there is no point to have any client creation logic in a cmdlet.  In each cmdlet you should just do $ServiceClient = Get-ServiceClient and leave all logic for creating the client in the function Get-ServiceClient (or whatever you want to call it).

As of now every cmdlet performs the same logic for client creation and I see no point in replicating this code in each cmdlet.  I manually remove all client creation logic and place in a single method and just call that method.

**Comment by deathly809** [comment]
In PSSwagger/PSSwagger.Constants.ps1.  That code will be copied to all cmdlets correct?  Is it possible to access those variables you set from GeneratedHelpers.ps1?  If so why not just move all that code into one spot?

**Comment by bmanikm** [comment]
@deathly809 Yes, those variables are accessible via New-ServiceClient utility function. Take a look at New-ServiceClient.ps1 file.

**Comment by deathly809** [comment]
@bmanikm What I don't understand is the code in PSSwagger.Constants.ps1.  Is that added to each cmdlet?  In other words are you just replicating code in each cmdlet that could be in one place?

**Pull request merged**